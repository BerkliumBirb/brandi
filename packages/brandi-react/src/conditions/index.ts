export * from './ConditionsContext';
export * from './ConditionsProvider';
export * from './tagged';
export * from './useConditions';
