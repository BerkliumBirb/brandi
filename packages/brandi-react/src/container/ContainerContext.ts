import { Container } from 'brandi';
import React from 'react';

export const ContainerContext = React.createContext<Container | null>(null);
