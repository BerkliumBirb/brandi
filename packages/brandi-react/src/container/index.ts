export * from './ContainerContext';
export * from './ContainerProvider';
export * from './useContainer';
