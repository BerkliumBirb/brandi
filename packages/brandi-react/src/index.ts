export { ContainerProvider } from './container';
export { createInjectionHooks, useInjection } from './injection';
export { tagged } from './tagged';
