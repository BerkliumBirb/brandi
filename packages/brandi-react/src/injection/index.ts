export * from './createInjectionHooks';
export * from './useInjection';
