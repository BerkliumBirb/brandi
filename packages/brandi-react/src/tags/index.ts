export * from './tagged';
export * from './TagsContext';
export * from './TagsProvider';
export * from './useTags';
