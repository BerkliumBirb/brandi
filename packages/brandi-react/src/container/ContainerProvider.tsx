import { Container } from 'brandi';
import React from 'react';

import { ContainerContext } from './ContainerContext';

export const ContainerProvider: React.FunctionComponent<{
  container: Container;
  cloning?: boolean;
}> = ({ children, container, cloning = false }) => {
  const resolvedContainer = React.useMemo(
    () => (cloning ? container.clone() : container),
    [cloning, container],
  );

  return (
    <ContainerContext.Provider value={resolvedContainer}>
      {children}
    </ContainerContext.Provider>
  );
};
