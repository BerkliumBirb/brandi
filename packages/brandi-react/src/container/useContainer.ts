import { Container } from 'brandi';
import React from 'react';

import { ContainerContext } from './ContainerContext';

export const useContainer = (): Container => {
  const container = React.useContext(ContainerContext);

  if (container === null || container === undefined) {
    throw new Error(
      "Could not get a container from a context. Did you forget to pass the container through 'ContainerProvider'?",
    );
  }

  return container;
};
