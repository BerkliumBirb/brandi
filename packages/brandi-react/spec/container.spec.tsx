import { Container, createContainer, token } from 'brandi';
import React from 'react';
import { renderHook } from '@testing-library/react-hooks';

import { ContainerProvider } from '../src';
import { useContainer } from '../src/container';

describe('container', () => {
  it("passes a container through 'ContainerProvider'", () => {
    const container = createContainer();

    const wrapper: React.FunctionComponent = ({ children }) => (
      <ContainerProvider container={container}>{children}</ContainerProvider>
    );

    const { result } = renderHook(() => useContainer(), { wrapper });

    expect(result.current).toBe(container);
  });

  it("passes a container clone through 'ContainerProvider' with 'cloning' prop", () => {
    const container = createContainer();

    const tokens = {
      some: token<number>('some'),
    };

    container.bind(tokens.some).toConstant(1);

    const wrapper: React.FunctionComponent = ({ children }) => (
      <ContainerProvider container={container} cloning>
        {children}
      </ContainerProvider>
    );

    const { result } = renderHook(() => useContainer(), { wrapper });

    container.bind(tokens.some).toConstant(2);

    expect(result.current).not.toBe(container);
    expect(result.current).toBeInstanceOf(Container);
    expect(result.current.get(tokens.some)).toBe(1);
  });

  it("throws error when a container is not passed through 'ContainerProvider'", () => {
    const { result } = renderHook(() => useContainer());

    expect(result.error).toBeInstanceOf(Error);
    expect(result.error).toMatchSnapshot();
  });
});
