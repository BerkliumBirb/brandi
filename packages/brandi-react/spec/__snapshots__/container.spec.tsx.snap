// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container throws error when a container is not passed through 'ContainerProvider' 1`] = `[Error: Could not get a container from a context. Did you forget to pass the container through 'ContainerProvider'?]`;
